src/components/LifeEventPicker.tsx

```diff
@@ -1,54 +1,287 @@
+    setState({
+      birth: !state.birth,
+      residence: state.residence,
+      marriage: state.marriage,
+      death: state.death,
+      anyEvent: state.anyEvent,
+    });
+  };
+
+  const addResidenceTextField = (): any => {
+    setState({
+      birth: state.birth,
+      residence: !state.residence,
+      marriage: state.marriage,
+      death: state.death,
+      anyEvent: state.anyEvent,
+    });
+  };
+
+  const addMarriageTextField = (): any => {
+    setState({
+      birth: state.birth,
+      residence: state.residence,
+      marriage: !state.marriage,
+      death: state.death,
+      anyEvent: state.anyEvent,
+    });
+  };
+
+  const addDeathTextField = (): any => {
+    setState({
+      birth: state.birth,
+      residence: state.residence,
+      marriage: state.marriage,
+      death: !state.death,
+      anyEvent: state.anyEvent,
+    });
+  };
+
+  const addAnyEventTextField = (): any => {
+    setState({
+      birth: state.birth,
+      residence: state.residence,
+      marriage: state.marriage,
+      death: state.death,
+      anyEvent: !state.anyEvent,
+    });
+  };
+
   return (
     <div className={classes.paper}>
       <Typography component="h1" variant="h6">
         Search with a life event
       </Typography>
-      <Typography style={{ marginBottom: 20, color: 'blue' }}>
-        Birth | Residence | Marriage | Death | Any
+      <Typography style={{ marginBottom: 20 }}>
+        <Link onClick={addBirthTextField}>Birth</Link>
+        {' | '}
+        <Link onClick={addResidenceTextField}>Residence</Link>
+        {' | '}
+        <Link onClick={addMarriageTextField}>Marriage</Link>
+        {' | '}
+        <Link onClick={addDeathTextField}>Death</Link>
+        {' | '}
+        <Link onClick={addAnyEventTextField}>Any</Link>
       </Typography>
-      <Grid container spacing={2}>
-        <Grid item xs={12} sm={6}>
-          <TextField
-            autoComplete="fname"
-            name="birthplace"
-            variant="outlined"
-            required
-            fullWidth
-            id="birthplace"
-            label="Birthplace"
-            autoFocus
-          />
-        </Grid>
-        <Grid item xs={6} sm={3}>
-          <TextField
-            variant="outlined"
-            required
-            fullWidth
-            id="startYear"
-            label="Year Start"
-            name="startYear"
-          />
-        </Grid>
-        <Grid item xs={6} sm={3}>
-          <TextField
-            variant="outlined"
-            required
-            fullWidth
-            id="endYear"
-            label="Year End"
-            name="endYear"
-          />
-        </Grid>
-      </Grid>
+
+      <div>
+        {state.birth ? (
+          <Grid container spacing={2} style={{ marginBottom: 20 }}>
+            <Grid item xs={12} sm={6}>
+              <TextField
+                name="birthplace"
+                variant="outlined"
+                required
```

Comment:
These fields aren't required, are they?

Reply:
I can remove name and required, but variant is needed to outline the text field :)

Reply:
Sorry, my comment wasn't clear! I meant only the 'required' property. I was trying to point out that all the TextFields in this form are not required in the sense that the user doesn't have to fill the out.

I believe 'name' will be needed to process the values when the form is submitted.